### PR TITLE
lbuga/ottc 215 ottc support new vici api for signaling information

### PR DESCRIPTION
**Pull request by lbuga-dn**

### CODE REVIEW

File: src/libcharon/plugins/vici/vici_control.c [hunk 1]
```diff
@@ -689,6 +689,35 @@ CALLBACK(reload_settings, vici_message_t*,
 	return send_reply(this, "reloading '%s' failed", lib->conf);
 }
 
+CALLBACK(activate_sa, vici_message_t*,
+	private_vici_control_t *this, char *name, u_int id, vici_message_t *request)
+{
+	int ike_id = request->get_int(request, -1, "ike-id");
```
Comment (lschlesinger-dn) [inline src/libcharon/plugins/vici/vici_control.c:695]: ike_sa IDs are unsigned integers

File: src/libcharon/plugins/vici/vici_control.c [hunk 2]
```diff
@@ -689,6 +689,35 @@ CALLBACK(reload_settings, vici_message_t*,
...
+	if (-1 == ike_id)
```
Comment (lschlesinger-dn) [inline src/libcharon/plugins/vici/vici_control.c:696]: -1 (= UINT32_MAX) is a valid ike_sa ID. Change to 0 which is not

File: src/libcharon/plugins/vici/vici_control.c [hunk 3]
```diff
@@ -689,6 +689,35 @@ CALLBACK(reload_settings, vici_message_t*,
...
+		return send_reply(this, "activating IKE_SA failed - missing 'ike-id'");
+
+	enumerator_t* isas = charon->controller->create_ike_sa_enumerator(
+		charon->controller, /* block */TRUE);
+
+	ike_sa_t *ike_sa;
+	bool found = FALSE;
+	while (isas->enumerate(isas, &ike_sa))
```
Comment (lschlesinger-dn) [inline src/libcharon/plugins/vici/vici_control.c:704]: Can replace this loop with `charon->ike_sa_manager->checkout_by_id()`. Don't forget to `checkin()` it afterwards.

File: src/libcharon/plugins/vici/vici_control.c [hunk 4]
```diff
@@ -689,6 +689,35 @@ CALLBACK(reload_settings, vici_message_t*,
...
+		// 				and `standby` for all others
+		if (ike_id != ike_sa->get_unique_id(ike_sa))
+		{
+			continue;
+		}
+		found = TRUE;
+	}
+
+	isas->destroy(isas);
+	char* ret = found ? NULL : "activating IKE_SA failed - unknown id";
```
Comment (lschlesinger-dn) [inline src/libcharon/plugins/vici/vici_control.c:716]: String literals are `const char*`

File: src/libcharon/plugins/vici/vici_control.c (hunk 3 above)
Comment (lbuga-dn) [reply src/libcharon/plugins/vici/vici_control.c:704]: I will still have to go over the rest of `ike_sa`s to signal standby for all except the one to activate.
So what will be the benefit of `checkout_by_id()`?

Comment (lschlesinger-dn) [reply src/libcharon/plugins/vici/vici_control.c:704]: I was commenting on the code as it is now, which only looks for a specific IKE SA. If you have changed planned in the future then you can ignore my comment.

**Review by lschlesinger-dn — APPROVED** [review]

**Pull request merged**